File: longest_common_prefix.py
"""
https://leetcode.com/problems/longest-common-prefix/description/
"""

from typing import List


class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        prefix = strs[0]

        for word in strs[1:]:
            while not word[0 : len(prefix)].startswith(prefix):
                prefix = prefix[:-1]

        return prefix

File: longestCommonPrefix.ts
// https://leetcode.com/problems/longest-common-prefix/description/

function longestCommonPrefix(strs: string[]): string {
    let prefix = strs[0];
    for (let i = 1; i < strs.length; i++) {
        while (!strs[i].slice(0, prefix.length).startsWith(prefix)) {
            prefix = prefix.slice(0,-1)
        }
    }
    return prefix
};
